test/remote/gateways/remote_hi_pay_test.rb

```diff
@@ -0,0 +1,85 @@
+require 'test_helper'
+
+class RemoteHiPayTest < Test::Unit::TestCase
+  def setup
+    @gateway = HiPayGateway.new(fixtures(:hi_pay))
+
+    @amount = 500
+    @credit_card = credit_card('4111111111111111', verification_value: '514', first_name: 'John', last_name: 'Smith', month: 12, year: 2025)
+
+    @options = {
+      order_id: "Sp_ORDER_#{SecureRandom.random_number(1000000000)}",
+      description: 'An authorize',
+      email: 'john.smith@test.com'
+    }
+
+    @billing_address = {
+      company: 'testCompany',
+         address1: '1 Main St.',
+         city: 'Burlington',
+         state: 'MA',
+         country: 'USA',
+         zip: '01803-3747',
+         phone: '1234567890'
+    }
+  end
+
+  def test_successful_authorize
+    response = @gateway.authorize(@amount, @credit_card, @options)
+
+    assert_success response
+    assert_equal response.message, 'Authorized'
+  end
+
+  def test_successful_purchase
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+    assert_include 'Captured', response.message
+  end
+
+  def test_successful_purchase_with_billing_address
```

Comment:
💬 Comment

Well this is a false positive, address is not being included and test pases

Reply:
Included thanks